web-bundle/src/main/java/com/graphhopper/swl/StableIdEncodedValues.java

```diff
@@ -43,17 +46,39 @@ final String getStableId(boolean reverse, EdgeIteratorState edge) {
         return Long.toUnsignedString(Longs.fromByteArray(stableId));
     }
 
-    final void setStableId(boolean reverse, EdgeIteratorState edge, String stableId) {
-        long stableIdLong = Long.parseUnsignedLong(stableId);
-        byte[] stableIdBytes = Longs.toByteArray(stableIdLong);
-
-        if (stableIdBytes.length != 8)
-            throw new IllegalArgumentException("stable ID must be 8 bytes: " + stableId);
+    public final void setStableId(boolean reverse, EdgeIteratorState edge) {
+        byte[] stableId = calculateStableEdgeId(edge);
+        if (stableId.length != 8)
+            throw new IllegalArgumentException("stable ID must be 8 bytes: " + new String(stableId));
 
         UnsignedIntEncodedValue[] idBytes = reverse ? reverseStableIdEnc : stableIdEnc;
         for (int i=0; i<8; i++) {
-            edge.set(idBytes[i], Byte.toUnsignedInt(stableIdBytes[i]));
+            edge.set(idBytes[i], Byte.toUnsignedInt(stableId[i]));
         }
     }
 
+    private static byte[] calculateStableEdgeId(EdgeIteratorState edge) {
+        // todo: how to get type of street?
+        String hashString = "Reference";
+
+        PointList geometry = edge.fetchWayGeometry(FetchMode.ALL);
+        GHPoint3D baseNode = geometry.get(edge.getBaseNode());
+        GHPoint3D adjNode = geometry.get(edge.getAdjNode());
+        double startLat = baseNode.getLat();
+        double startLong = baseNode.getLon();
+        double endLat = adjNode.getLat();
+        double endLong = adjNode.getLon();
+        hashString += String.format(" %.5f %.5f %.5f %.5f", startLong, startLat, endLong, endLat);
```

Comment:
Do you remember what precision we write csv at? Are the hashes consistent?

Reply:
It looks like in the street edges CSVs we write lat/longs with 6 places after the decimal...and [we hash using the same number as well](https://github.com/replicahq/model/blob/master/mapservice/api/network_link_references.py#L58-L73)

However, the [original shared streets implementation](https://github.com/sharedstreets/sharedstreets-builder/blob/master/src/main/java/io/sharedstreets/data/SharedStreetsReference.java#L331-L346) only uses 5 places, which is why I chose this I think. Should I switch it to 6 to keep it consistent with our current hashing scheme?